lib/entities/quote/RelayQuote.ts

```diff
@@ -0,0 +1,304 @@
+  public static fromClassicQuote(request: RelayRequest, classicQuote: ClassicQuote): RelayQuote {
+    // Relay quotes require a gas token estimation
+    if (!classicQuote.gasUseEstimateGasToken) {
+      throw new Error('Classic quote must have gasUseEstimateGasToken');
+    }
+    const gasEstimateInFeeToken = request.config.feeAmountStartOverride
+      ? BigNumber.from(request.config.feeAmountStartOverride)
+      : classicQuote.gasUseEstimateGasToken;
+    const {
+      feeAmountStart,
+      feeAmountEnd,
+    } = this.applyGasAdjustment(gasEstimateInFeeToken, classicQuote);
+
+    return new RelayQuote({
+      createdAtMs: classicQuote.createdAtMs,
+      request,
+      chainId: request.info.tokenInChainId,
+      requestId: request.info.requestId,
+      quoteId: classicQuote.quoteId,
+      tokenIn: request.info.tokenIn,
+      tokenOut: request.info.tokenOut,
+      amountIn: classicQuote.amountIn, // apply no gas adjustment
+      amountOut: classicQuote.amountOut, // apply no gas adjustment
+      feeAmountStart,
+      feeAmountEnd,
+      swapper: request.config.swapper,
+      classicQuoteData: classicQuote.toJSON(),
+      nonce: generateRandomNonce(), // add a nonce
+    });
+  }
+
+  private constructor(args: RelayQuoteConstructorArgs) {
+    Object.assign(this, args, {
+      createdAtMs: args.createdAtMs || currentTimestampInMs(),
+    });
+    this.createdAt = timestampInMstoSeconds(parseInt(this.createdAtMs));
+    // regenerate the classic quote from the data
+    this.classicQuote = ClassicQuote.fromResponseBody(this.request, this.classicQuoteData);
+  }
+
+  public toJSON(): RelayQuoteDataJSON {
+    return {
+      orderInfo: this.toOrder().toJSON(),
+      encodedOrder: this.toOrder().serialize(),
+      quoteId: this.quoteId,
+      requestId: this.requestId,
+      orderHash: this.toOrder().hash(),
+      startTimeBufferSecs: this.startTimeBufferSecs,
+      auctionPeriodSecs: this.auctionPeriodSecs,
+      deadlineBufferSecs: this.deadlineBufferSecs,
+      slippageTolerance: this.request.info.slippageTolerance,
+      permitData: this.getPermitData(),
+      classicQuoteData: this.classicQuoteData,
+    };
+  }
+
+  public toOrder(): RelayOrder {
+    const orderBuilder = new RelayOrderBuilder(this.chainId);
+    const feeStartTime = Math.floor(Date.now() / 1000);
+    const deadline = feeStartTime + this.auctionPeriodSecs + this.deadlineBufferSecs;
+    const nonce = this.nonce ?? generateRandomNonce();
+
+    const builder = orderBuilder
+      .swapper(ethers.utils.getAddress(this.request.config.swapper))
+      .nonce(BigNumber.from(nonce))
+      .deadline(deadline)
+      // Add the swap input to UR
+      .input({
+        token: this.tokenIn,
+        amount: this.amountIn,
+        recipient: UNIVERSAL_ROUTER_ADDRESS(this.chainId),
+      })
+      // Add the gas token input to the filler
+      .fee({
+        token: this.request.config.gasToken,
```

Comment:
Do we need to do anything special for FOT tokens or will these likely not be supported by fillers?

Reply:
Oh good question, so fee on transfer should be handled natively since fillers will have to price that into their filling strategies. Quoting onchain should reflect any buy/sell taxes too